src/server/generic_family.cc

```diff
@@ -28,6 +31,59 @@ using namespace std;
 using namespace facade;
 
 namespace {
+using VersionBuffer = std::array<char, 2>;
+using CrcBuffer = std::array<char, sizeof(uint64_t)>;
+
+VersionBuffer MakeRdbVersion() {
+  VersionBuffer buf;
+  buf[0] = RDB_VERSION & 0xff;
+  buf[1] = (RDB_VERSION >> 8) & 0xff;
+  return buf;
+}
+
+CrcBuffer MakeCheckSum(std::string_view dump_res) {
+  uint64_t chksum = crc64(0, reinterpret_cast<const uint8_t*>(dump_res.data()), dump_res.size());
+  CrcBuffer buf;
+  absl::little_endian::Store64(buf.data(), chksum);
+  return buf;
+}
+
+std::string AppendFooter(std::string_view dump_res) {
+  /* Write the footer, this is how it looks like:
+   * ----------------+---------------------+---------------+
+   * ... RDB payload | 2 bytes RDB version | 8 bytes CRC64 |
+   * ----------------+---------------------+---------------+
+   * RDB version and CRC are both in little endian.
+   */
+  const auto ver = MakeRdbVersion();
+  std::string complete_message(dump_res);
+  complete_message.append(ver.data(), ver.size());
+  const auto crc = MakeCheckSum(complete_message);
+  complete_message.append(crc.data(), crc.size());
+  return complete_message;
+}
+
+class BytesSink : public io::Sink {
```

Comment:
`io::StringSink` in `io.h` does exactly this :)

Reply:
I look up for any previous impl for this, failed to find it, I will change this